apps/web/src/components/PoolCandidatesTable/JobPlacementDialog.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useState } from "react";
+import { useIntl } from "react-intl";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useMutation } from "urql";
+import MapPinIcon from "@heroicons/react/24/outline/MapPinIcon";
+
+import {
+  RadioGroup,
+  Select,
+  Submit,
+  enumToOptions,
+  objectsToSortedOptions,
+} from "@gc-digital-talent/forms";
+import {
+  Department,
+  FragmentType,
+  Maybe,
+  PlacementType,
+  PoolCandidateStatus,
+  getFragment,
+  graphql,
+  makeFragmentData,
+} from "@gc-digital-talent/graphql";
+import {
+  commonMessages,
+  errorMessages,
+  getPlacementType,
+} from "@gc-digital-talent/i18n";
+import { Button, Dialog } from "@gc-digital-talent/ui";
+import { toast } from "@gc-digital-talent/toast";
+
+import { isQualifiedStatus, statusToJobPlacement } from "~/utils/poolCandidate";
+import poolCandidateMessages from "~/messages/poolCandidateMessages";
+
+export const PLACEMENT_TYPE_STATUSES = [
+  PoolCandidateStatus.PlacedCasual,
+  PoolCandidateStatus.PlacedIndeterminate,
+  PoolCandidateStatus.PlacedTentative,
+  PoolCandidateStatus.PlacedTerm,
+];
+
+const PlaceCandidate_Mutation = graphql(/* GraphQL */ `
+  mutation PlaceCandidate_Mutation(
+    $id: UUID!
+    $placeCandidate: PlaceCandidateInput!
+  ) {
+    placeCandidate(id: $id, placeCandidate: $placeCandidate) {
+      id
+    }
+  }
+`);
+
+const RevertPlaceCandidate_Mutation = graphql(/* GraphQL */ `
+  mutation RevertPlaceCandidate_Mutation($id: UUID!) {
+    revertPlaceCandidate(id: $id) {
+      id
+    }
+  }
+`);
+
+export const JobPlacementDialog_Fragment = graphql(/* GraphQL */ `
+  fragment JobPlacementDialog_Fragment on PoolCandidate {
+    id
+    status
+    placedDepartment {
+      id
+      name {
+        en
+        fr
+      }
+    }
+  }
+`);
+
+type FormValues = {
+  placedJob?: PlacementType | "NOT_PLACED";
+  placedDepartment?: string;
+};
+
+interface JobPlacementDialogProps {
+  jobPlacementDialogQuery: FragmentType<typeof JobPlacementDialog_Fragment>;
+  departments: Department[];
+  context?: "table" | "view";
+}
+
+export const JobPlacementDialog = ({
+  jobPlacementDialogQuery,
+  departments,
+  context = "table",
+}: JobPlacementDialogProps) => {
+  const intl = useIntl();
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [, executePlacedCandidate] = useMutation(PlaceCandidate_Mutation);
+  const [, executeRevertPlacedCandidate] = useMutation(
+    RevertPlaceCandidate_Mutation,
+  );
+
+  const {
+    id: poolCandidateId,
+    status,
+    placedDepartment,
+  } = getFragment(JobPlacementDialog_Fragment, jobPlacementDialogQuery);
+
+  const placedJob =
+    status && PLACEMENT_TYPE_STATUSES.includes(status)
+      ? (status as unknown as PlacementType)
+      : "NOT_PLACED";
```

Comment:
Okay fixed by [b6fe696](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10216/commits/b6fe6960a81246f767b6c22f5b295a37fd0333a6). I just opted to using the PLACEMENT_TYPE_STATUSES since it includes placed tentatively.